mixedbread_ai_langchain/retrievers/vector_store_file_retriever.py

```diff
@@ -0,0 +1,326 @@
+                max_retries=max_retries,
+            )
+
+        # Use unified client approach - async client is available via MixedbreadClient
+        # No separate async client management needed
+
+    def _convert_file_results_to_documents(self, search_response) -> List[Document]:
+        """Convert Mixedbread file search results to LangChain Documents."""
+        documents = []
+
+        for file_result in search_response.data:
+            page_content = ""
+
+            if hasattr(file_result, "chunks") and file_result.chunks:
+                chunk_texts = []
+                for chunk in file_result.chunks:
+                    if hasattr(chunk, "content") and chunk.content:
+                        chunk_texts.append(chunk.content)
+
+                if self.chunk_limit and len(chunk_texts) > self.chunk_limit:
+                    chunk_texts = chunk_texts[: self.chunk_limit]
+
+                page_content = "\n\n".join(chunk_texts)
+
+            if not page_content:
+                page_content = (
+                    f"File: {getattr(file_result, 'filename', 'Unknown file')}"
+                )
+
+            metadata = {
+                "source": getattr(file_result, "filename", "unknown"),
+                "file_id": getattr(file_result, "id", None),
+                "filename": getattr(file_result, "filename", None),
+                "vector_store_id": getattr(file_result, "vector_store_id", None),
+                "relevance_score": getattr(file_result, "score", 0.0),
+                "status": getattr(file_result, "status", None),
+                "usage_bytes": getattr(file_result, "usage_bytes", None),
+                "created_at": getattr(file_result, "created_at", None),
+                "version": getattr(file_result, "version", None),
+                "vector_store_file_retrieval": True,
+                "file_based_search": True,
+            }
+
+            if hasattr(file_result, "metadata") and file_result.metadata:
+                if isinstance(file_result.metadata, dict):
+                    metadata["file_metadata"] = file_result.metadata
+
+            if hasattr(file_result, "chunks") and file_result.chunks:
+                metadata["total_chunks"] = len(file_result.chunks)
+                metadata["chunks_included"] = min(
+                    len(file_result.chunks), self.chunk_limit or len(file_result.chunks)
+                )
+
+                chunk_details = []
+                for i, chunk in enumerate(file_result.chunks):
+                    if self.chunk_limit and i >= self.chunk_limit:
+                        break
+
+                    chunk_info = {
+                        "position": getattr(chunk, "position", i),
+                        "score": getattr(chunk, "score", 0.0),
+                        "file_id": getattr(chunk, "file_id", None),
+                        "vector_store_id": getattr(chunk, "vector_store_id", None),
+                    }
```

Comment:
What about the multi-modal chunk fields like test or ocr_text?

Reply:
I am not even sure why we need to construct a new dict here and in other places as well. Why not use `file_results.chunks` directly?